source/beamline/CLS/CLSSIS3820ScalerContinuousMode.cpp

```diff
@@ -0,0 +1,192 @@
+#include "CLSSIS3820ScalerContinuousMode.h"
+#include "beamline/CLS/CLSSIS3820Scaler.h"
+#include "actions3/AMActionSupport.h"
+#include "actions3/AMListAction3.h"
+
+CLSSIS3820ScalerContinuousMode::CLSSIS3820ScalerContinuousMode(const QString &name, QObject *parent) :
+	AMEnumeratedControl(name, "", parent)
+{
+	// Initialize inherited variables.
+
+	setAllowsMovesWhileMoving(false);
+	setContextKnownDescription("Mode Control");
+
+	// Initialize local variables.
+
+	scanCountControl_ = 0;
+	numberOfScansPerBufferControl_ = 0;
+	startScanControl_ = 0;
+
+	singleShotScanCountValue_ = 1;
+	singleShotNumberOfScansPerBufferValue_ = 1;
+
+	// Setup value options.
+
+	addOption(CLSSIS3820Scaler::SingleShot, "Single shot");
+	addOption(CLSSIS3820Scaler::Continuous, "Continuous");
+}
+
+CLSSIS3820ScalerContinuousMode::~CLSSIS3820ScalerContinuousMode()
+{
+
+}
+
+bool CLSSIS3820ScalerContinuousMode::canMeasure() const
+{
+	bool result = false;
+
+	if (isConnected())
+		result = scanCountControl_->canMeasure() && numberOfScansPerBufferControl_->canMeasure() && startScanControl_->canMeasure();
+
+	return result;
+}
+
+bool CLSSIS3820ScalerContinuousMode::canMove() const
+{
+	bool result = false;
+
+	if (isConnected())
+		result = scanCountControl_->canMove() && numberOfScansPerBufferControl_->canMove() && startScanControl_->canMove();
+
+	return result;
+}
+
+bool CLSSIS3820ScalerContinuousMode::isContinuous() const
+{
+	bool result = false;
+
+	if (isConnected()) {
+		if (startScanControl_->value() == CLSSIS3820Scaler::Scanning && scanCountControl_->value() == 0) {
+			result = true;
+		}
+	}
+
+	return result;
+}
+
+bool CLSSIS3820ScalerContinuousMode::isSingleShot() const
+{
+	bool result = false;
+
+	if (isConnected()) {
+		if (scanCountControl_->value() > 0)
+			result = true;
+	}
+
+	return result;
+}
+
```

Comment:
That is definitely a good point.  Since we have very rigid definitions of what it means to be in single shot or continuous, we may as well have an Unknown state for when that is not the case.